packages/@coorpacademy-app-review/src/index.tsx

```diff
@@ -34,59 +33,35 @@ const ConnectedApp = (options: ConnectedOptions): JSX.Element => {
   return <AppReviewTemplate {...props} />;
 };
 
+const storeTokenAndCreateProgression = async (
+  store: Store<StoreState>,
+  options: AppOptions
+): Promise<void> => {
+  const token = get('token', options);
+  if (store === null || isEmpty(token)) return;
+  store.dispatch(storeToken(token));
+
+  const skillRef = get('skillRef', options);
+
+  if (skillRef) {
+    store.dispatch(navigateTo('loader'));
+    await store.dispatch(postProgression(skillRef));
+    store.dispatch(navigateTo('slides'));
+  }
+};
+
 const AppReview = ({options}: {options: AppOptions}): JSX.Element | null => {
   const [store, setStore] = useState<Store<StoreState, AnyAction> | null>(null);
-  const [isProgressionCreated, setIsProgressionCreated] = useState(false);
-
   const {translate, onQuitClick, skin} = options;
 
   useEffect(() => {
-    if (store) return;
-
     const newStore = configureStore(options);
     setStore(newStore);
-  }, [options, store]);
-
-  useEffect(() => {
-    if (!store || isProgressionCreated) return;
-
-    return store.subscribe(() => {
-      const isProgressionPresent = !isEmpty(store.getState().data.progression);
-      return isProgressionPresent && setIsProgressionCreated(isProgressionPresent);
-    });
-  }, [isProgressionCreated, options, store]);
-
-  useEffect(() => {
-    const token = get('token', options);
-    if (store === null || isEmpty(token)) return;
-
-    store.dispatch(storeToken(token));
-  }, [options, store]);
-
-  useEffect(() => {
-    const token = get('token', options);
-    if (store === null || isEmpty(token)) return;
-
-    const skillRef = get('skillRef', options);
-
-    /* ThunkAction is not assignable to parameter of type 'AnyAction'
-      ts problem is described here = https://github.com/reduxjs/redux-thunk/issues/333 */
-    skillRef ? store.dispatch(postProgression(skillRef)) : store.dispatch(fetchSkills);
```

Comment:
i dont see any part about fetch skills anymore is it not required anymore here ?

Reply:
yes: we'll remove all `skills` parts since it will eventually be implemented on mooc + mobile sides.
`app-review` focuses 100% on the shared redux implementation for the slides "player" only.